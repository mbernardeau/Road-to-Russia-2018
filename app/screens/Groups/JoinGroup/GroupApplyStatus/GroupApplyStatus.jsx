import React from 'react'
import PropTypes from 'prop-types'

import Snackbar from 'material-ui/Snackbar'
import IconButton from 'material-ui/IconButton'
import CloseIcon from '@material-ui/icons/Close'

const GroupApplyStatus = ({ status, name, reason, handleClose }) => (
  <Snackbar
    anchorOrigin={{
      vertical: 'bottom',
      horizontal: 'left',
    }}
    open={!!status}
    autoHideDuration={6000}
    onClose={handleClose}
    SnackbarContentProps={{
      'aria-describedby': 'message-id',
    }}
    message={
      <span id="message-id">
        {status === 'failed' ? reason : `Demande envoy√©e pour la tribu ${name}`}
      </span>
    }
    action={[
      <IconButton key="close" aria-label="Close" color="inherit" onClick={handleClose}>
        <CloseIcon />
      </IconButton>,
    ]}
  />
)

GroupApplyStatus.propTypes = {
  status: PropTypes.string,
  name: PropTypes.string,
  reason: PropTypes.string,
  handleClose: PropTypes.func.isRequired,
}

export default GroupApplyStatus
