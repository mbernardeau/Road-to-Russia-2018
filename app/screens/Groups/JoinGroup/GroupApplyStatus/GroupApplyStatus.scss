.group-join-link {
  color: #7f7ff6;
}
