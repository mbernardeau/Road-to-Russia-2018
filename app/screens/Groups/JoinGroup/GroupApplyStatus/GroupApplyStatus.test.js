import React from 'react'
import renderer from 'react-test-renderer'

import GroupApplyStatus from './GroupApplyStatus'

jest.mock('@material-ui/core/Snackbar', () => 'Snackbar')
jest.mock('@material-ui/core/IconButton', () => 'IconButton')
jest.mock('@material-ui/icons/Close', () => 'CloseIcon')

describe('screens/JoinGroup/GroupApplyOkSnackbar', () => {
  it('renders correctly without props', () => {
    const props = {
      handleClose: jest.fn(),
    }

    const tree = renderer.create(<GroupApplyStatus {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly when failed props', () => {
    const props = {
      handleClose: jest.fn(),
      status: 'failed',
      reason: "ma super tribu n'existe pas",
    }

    const tree = renderer.create(<GroupApplyStatus {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly when success', () => {
    const props = {
      handleClose: jest.fn(),
      status: 'success',
      name: 'ma super tribu',
    }

    const tree = renderer.create(<GroupApplyStatus {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
