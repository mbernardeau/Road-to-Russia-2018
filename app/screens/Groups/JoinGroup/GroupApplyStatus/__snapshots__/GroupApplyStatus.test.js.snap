// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screens/JoinGroup/GroupApplyOkSnackbar renders correctly when failed props 1`] = `
<Snackbar
  SnackbarContentProps={
    Object {
      "aria-describedby": "message-id",
      "className": "group-apply-snackbar",
    }
  }
  action={
    Array [
      <IconButton
        aria-label="Close"
        color="inherit"
        onClick={[MockFunction]}
      >
        <CloseIcon />
      </IconButton>,
    ]
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={60000}
  message={
    <span
      id="message-id"
    >
      ma super tribu n'existe pas
    </span>
  }
  onClose={[MockFunction]}
  open={true}
/>
`;

exports[`screens/JoinGroup/GroupApplyOkSnackbar renders correctly when success 1`] = `
<Snackbar
  SnackbarContentProps={
    Object {
      "aria-describedby": "message-id",
      "className": "group-apply-snackbar",
    }
  }
  action={
    Array [
      <IconButton
        aria-label="Close"
        color="inherit"
        onClick={[MockFunction]}
      >
        <CloseIcon />
      </IconButton>,
    ]
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={60000}
  message={
    <span
      id="message-id"
    >
      <React.Fragment>
        Demande envoyée pour la tribu 
        <b />
         !
         
        <React.Fragment>
          Payer 
          € sur la
           
          <a
            className="group-join-link"
            href="https://www.paypal.com/pools/c/84gsKV8QG8"
            target="_blank"
            title="Site cagnotte"
          >
            cagnotte
          </a>
           
          pour valider votre inscription !
        </React.Fragment>
      </React.Fragment>
    </span>
  }
  onClose={[MockFunction]}
  open={true}
/>
`;

exports[`screens/JoinGroup/GroupApplyOkSnackbar renders correctly without props 1`] = `
<Snackbar
  SnackbarContentProps={
    Object {
      "aria-describedby": "message-id",
      "className": "group-apply-snackbar",
    }
  }
  action={
    Array [
      <IconButton
        aria-label="Close"
        color="inherit"
        onClick={[MockFunction]}
      >
        <CloseIcon />
      </IconButton>,
    ]
  }
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  autoHideDuration={60000}
  message={
    <span
      id="message-id"
    >
      <React.Fragment>
        Demande envoyée pour la tribu 
        <b />
         !
         
        <React.Fragment>
          Payer 
          € sur la
           
          <a
            className="group-join-link"
            href="https://www.paypal.com/pools/c/84gsKV8QG8"
            target="_blank"
            title="Site cagnotte"
          >
            cagnotte
          </a>
           
          pour valider votre inscription !
        </React.Fragment>
      </React.Fragment>
    </span>
  }
  onClose={[MockFunction]}
  open={false}
/>
`;
